diff --git a/src/plugins/jack/Makefile.am b/src/plugins/jack/Makefile.am
index c3c5173..27a3ee3 100644
--- a/src/plugins/jack/Makefile.am
+++ b/src/plugins/jack/Makefile.am
@@ -4,7 +4,7 @@ AUTOMAKE_OPTIONS = foreign
 
 if HAVE_JACK
 
-jackltlibs = libjack.la
+jackltlibs = libjackdriver.la
 jacksources = ao_jack.c ao_jack_resample.c
 
 else
@@ -19,9 +19,9 @@ AM_CPPFLAGS = -I$(top_builddir)/include/ao -I$(top_srcdir)/include
 libdir = $(plugindir)
 lib_LTLIBRARIES = $(jackltlibs)
 
-libjack_la_CFLAGS = @JACK_CFLAGS@
-libjack_la_LDFLAGS = @PLUGIN_LDFLAGS@ @JACK_LDFLAGS@
-libjack_la_LIBADD = @JACK_LIBS@ ../../libao.la
-libjack_la_SOURCES = $(jacksources)
+libjackdriver_la_CFLAGS = @JACK_CFLAGS@
+libjackdriver_la_LDFLAGS = @PLUGIN_LDFLAGS@ @JACK_LDFLAGS@
+libjackdriver_la_LIBADD = @JACK_LIBS@ ../../libao.la
+libjackdriver_la_SOURCES = $(jacksources)
 
 EXTRA_DIST = ao_jack.c
-- 
2.1.2

